
Traceback (most recent call last):
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\cbook.py", line 361, in process
    func(*args, **kwargs)
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\backend_bases.py", line 2954, in mouse_move
    self.set_message(self._mouse_event_to_message(event))
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\backends\_backend_tk.py", line 722, in set_message
    self.message.set(s)
  File "C:\Users\Dell\AppData\Local\Programs\Python\Python311\Lib\tkinter\__init__.py", line 424, in set
    return self._tk.globalsetvar(self._name, value)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: main thread is not in main loop
Traceback (most recent call last):
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\cbook.py", line 361, in process
    func(*args, **kwargs)
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\backend_bases.py", line 2954, in mouse_move
    self.set_message(self._mouse_event_to_message(event))
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\backends\_backend_tk.py", line 722, in set_message
    self.message.set(s)
  File "C:\Users\Dell\AppData\Local\Programs\Python\Python311\Lib\tkinter\__init__.py", line 424, in set
    return self._tk.globalsetvar(self._name, value)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: main thread is not in main loop
Traceback (most recent call last):
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\cbook.py", line 361, in process
    func(*args, **kwargs)
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\backend_bases.py", line 2954, in mouse_move
    self.set_message(self._mouse_event_to_message(event))
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\backends\_backend_tk.py", line 722, in set_message
    self.message.set(s)
  File "C:\Users\Dell\AppData\Local\Programs\Python\Python311\Lib\tkinter\__init__.py", line 424, in set
    return self._tk.globalsetvar(self._name, value)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: main thread is not in main loop
Traceback (most recent call last):
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\cbook.py", line 361, in process
    func(*args, **kwargs)
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\backend_bases.py", line 2954, in mouse_move
    self.set_message(self._mouse_event_to_message(event))
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\backends\_backend_tk.py", line 722, in set_message
    self.message.set(s)
  File "C:\Users\Dell\AppData\Local\Programs\Python\Python311\Lib\tkinter\__init__.py", line 424, in set
    return self._tk.globalsetvar(self._name, value)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: main thread is not in main loop
Traceback (most recent call last):
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\cbook.py", line 361, in process
    func(*args, **kwargs)
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\backend_bases.py", line 2954, in mouse_move
    self.set_message(self._mouse_event_to_message(event))
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\backends\_backend_tk.py", line 722, in set_message
    self.message.set(s)
  File "C:\Users\Dell\AppData\Local\Programs\Python\Python311\Lib\tkinter\__init__.py", line 424, in set
    return self._tk.globalsetvar(self._name, value)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: main thread is not in main loop
Traceback (most recent call last):
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\cbook.py", line 361, in process
    func(*args, **kwargs)
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\backend_bases.py", line 2954, in mouse_move
    self.set_message(self._mouse_event_to_message(event))
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\backends\_backend_tk.py", line 722, in set_message
    self.message.set(s)
  File "C:\Users\Dell\AppData\Local\Programs\Python\Python311\Lib\tkinter\__init__.py", line 424, in set
    return self._tk.globalsetvar(self._name, value)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: main thread is not in main loop
Traceback (most recent call last):
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\cbook.py", line 361, in process
    func(*args, **kwargs)
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\backend_bases.py", line 2954, in mouse_move
    self.set_message(self._mouse_event_to_message(event))
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\backends\_backend_tk.py", line 722, in set_message
    self.message.set(s)
  File "C:\Users\Dell\AppData\Local\Programs\Python\Python311\Lib\tkinter\__init__.py", line 424, in set
    return self._tk.globalsetvar(self._name, value)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: main thread is not in main loop
Traceback (most recent call last):
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\cbook.py", line 361, in process
    func(*args, **kwargs)
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\backend_bases.py", line 2954, in mouse_move
    self.set_message(self._mouse_event_to_message(event))
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\backends\_backend_tk.py", line 722, in set_message
    self.message.set(s)
  File "C:\Users\Dell\AppData\Local\Programs\Python\Python311\Lib\tkinter\__init__.py", line 424, in set
    return self._tk.globalsetvar(self._name, value)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: main thread is not in main loop
Traceback (most recent call last):
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\cbook.py", line 361, in process
    func(*args, **kwargs)
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\backend_bases.py", line 2954, in mouse_move
    self.set_message(self._mouse_event_to_message(event))
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\backends\_backend_tk.py", line 722, in set_message
    self.message.set(s)
  File "C:\Users\Dell\AppData\Local\Programs\Python\Python311\Lib\tkinter\__init__.py", line 424, in set
    return self._tk.globalsetvar(self._name, value)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: main thread is not in main loop
Traceback (most recent call last):
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\cbook.py", line 361, in process
    func(*args, **kwargs)
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\backend_bases.py", line 2954, in mouse_move
    self.set_message(self._mouse_event_to_message(event))
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\backends\_backend_tk.py", line 722, in set_message
    self.message.set(s)
  File "C:\Users\Dell\AppData\Local\Programs\Python\Python311\Lib\tkinter\__init__.py", line 424, in set
    return self._tk.globalsetvar(self._name, value)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: main thread is not in main loop
Traceback (most recent call last):
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\cbook.py", line 361, in process
    func(*args, **kwargs)
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\backend_bases.py", line 2954, in mouse_move
    self.set_message(self._mouse_event_to_message(event))
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\backends\_backend_tk.py", line 722, in set_message
    self.message.set(s)
  File "C:\Users\Dell\AppData\Local\Programs\Python\Python311\Lib\tkinter\__init__.py", line 424, in set
    return self._tk.globalsetvar(self._name, value)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: main thread is not in main loop
Traceback (most recent call last):
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\cbook.py", line 361, in process
    func(*args, **kwargs)
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\backend_bases.py", line 2954, in mouse_move
    self.set_message(self._mouse_event_to_message(event))
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\backends\_backend_tk.py", line 722, in set_message
    self.message.set(s)
  File "C:\Users\Dell\AppData\Local\Programs\Python\Python311\Lib\tkinter\__init__.py", line 424, in set
    return self._tk.globalsetvar(self._name, value)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: main thread is not in main loop
Traceback (most recent call last):
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\cbook.py", line 361, in process
    func(*args, **kwargs)
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\backend_bases.py", line 2954, in mouse_move
    self.set_message(self._mouse_event_to_message(event))
  File "C:\Users\Dell\OneDrive\Desktop\PROJECTS\COSTUMERS\TRADE_PATTERN\venv311\Lib\site-packages\matplotlib\backends\_backend_tk.py", line 722, in set_message
    self.message.set(s)